doc/releases/changelog-0.36.0.md

```diff
@@ -555,14 +669,14 @@
   and will be added again in the next release.
   [(#5612)](https://github.com/PennyLaneAI/pennylane/pull/5612)
 
-* Cast the keys of the `CountsMP` measurements returned `dynamic_one_shot` to the type produced by `MeasurementValue.concretize`.
+* `qml.counts` now returns the same keys with `dynamic_one_shot` and `defer_measurements`.
   [(#5587)](https://github.com/PennyLaneAI/pennylane/pull/5587)
 
 * `null.qubit` now automatically supports any operation without a decomposition.
   [(#5582)](https://github.com/PennyLaneAI/pennylane/pull/5582)
 
 * Fixed a bug where the shape and type of derivatives obtained by applying a gradient transform to
-  a QNode differed, based on whether the QNode uses classical coprocessing.
+  a QNode differed based on whether the QNode uses classical coprocessing.
   [(#4945)](https://github.com/PennyLaneAI/pennylane/pull/4945)
 
```

Comment:
Hey @vincentmr! Can you explain what's going on in https://github.com/PennyLaneAI/pennylane/pull/5529? I'm a little confused 😅 (this is line 699 in the changelog)

Reply:
Non-computational-basis measurements require applying diagonalization operators to the state. This is done via `rotations` on some device, but not Lightning. For some measurements (e.g. `expval`) the Lightning devices applies the diag operators itself, but not for `probs`, and this has to be done on the PennyLane side. This was not done previously, and hence non-computational-basis observables would lead to incorrect measurement results. This will all go away after the port to the new device API.